File: backend/services/post-rust-actix-web-fdm/src/common/api/invalid_text_post_content.rs
use crate::common::utils::error::*;

impl super::super::TextPostContent {
    pub(in crate::common) fn text_post_content_empty() -> ErrorBody {
        ErrorBody {
            error: ErrBody {
                error: "TEXT_POST_CONTENT_EMPTY".into(),
                reason: "text post content is empty".to_string(),
                message: "text post content is empty".to_string(),
            },
        }
    }
    pub(in crate::common) fn text_post_content_too_long() -> ErrorBody {
        ErrorBody {
            error: ErrBody {
                error: "TEXT_POST_CONTENT_TOO_LONG".into(),
                reason: "text post content is too long".to_string(),
                message: "text post content is too long".to_string(),
            },
        }
    }
}

File: backend/services/post-kt-vertx-fdm/src/main/kotlin/me/jason5lee/post_kt_vertx_fdm/common/api/InvalidTextPostContent.kt
package me.jason5lee.post_kt_vertx_fdm.common.api

import me.jason5lee.post_ktor_mongo_fdm.common.utils.Err
import me.jason5lee.post_ktor_mongo_fdm.common.utils.FailureBody

object InvalidTextPostContent {
    internal val empty: FailureBody = FailureBody(
        error = Err(
            error = "TEXT_POST_CONTENT_EMPTY",
            reason = "Text post content cannot be empty",
            message = "Text post content cannot be empty",
        )
    )
    internal val tooLong: FailureBody = FailureBody(
        error = Err(
            error = "TEXT_POST_CONTENT_TOO_LONG",
            reason = "Text post content must be at most 65535 characters",
            message = "Text post content must be at most 65535 characters",
        )
    )
}
